proto/proto/penumbra/core/transaction/v1alpha1/transaction.proto

```diff
@@ -41,6 +41,24 @@ message TransactionBody {
   // An optional encrypted memo. It will only be populated if there are
   // outputs in the actions of this transaction. 528 bytes.
   optional bytes encrypted_memo = 6;
+  // A unix timestamp, in nanoseconds.
+  // If this value is set, the transaction is valid iff the most recent block header
+  // timestamp is < this value.
+  //
+  // If the value is zero, this value is considered unset.
+  // [TODO: Double check that this actually does behave like google's protobufs]
+  // The behavior of this time stamp should match google's protobuf specification:
+  // [specification]: https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Timestamp
+  uint64 valid_before = 7;
```

Comment:
This is supposed to match exactly the Timestamp type in IBC, which has this note on it. I don't have the context to verify this yet, hence the TODO.